packages/common/src/util/helpers.ts

```diff
@@ -62,28 +80,67 @@ export const checkWebSocketHost = (host: string): string => {
  *
  * @returns Object with error | result key to identify success or fail
  */
-export const destructResponse = (response: any, nodeId: string = null): { [key: string]: any } => {
-  const { result = {}, error } = response
+export const destructResponse = (
+  response: any,
+  nodeId: string = null,
+): {[key: string]: any} => {
+  const {result = {}, error} = response
   if (error) {
-    return { error }
+    return {error}
   }
-  const { result: nestedResult = null } = result
+  const {result: nestedResult = null} = result
   if (nestedResult === null) {
     if (nodeId !== null) {
       result.node_id = nodeId
     }
-    return { result }
+    return {result}
   }
-  const { code = null, node_id = null, result: vertoResult = null } = nestedResult
+  const {code = null, node_id = null, result: vertoResult = null} = nestedResult
   if (code && code !== '200') {
-    return { error: nestedResult }
+    return {error: nestedResult}
   }
   if (vertoResult) {
     return destructResponse(vertoResult, node_id)
   }
-  return { result: nestedResult }
+  return {result: nestedResult}
 }
 
 export const randomInt = (min: number, max: number) => {
   return Math.floor(Math.random() * (max - min + 1) + min)
 }
+
+
+// Build a proxy intended to normalize RTCPeerConnection implementations
+// by forcing the usage of alternative async methods with the suffix `asyncAlternativeSuffix`
+// and making `promisifyTargetMethods` to always return Promises
+export const normalizeAsyncAPIs = <T extends object>(
```

Comment:
This is required since the Citrix implementation deals with remote requests to the local machine.
In a traditional WebRTC implementation, the same requests are `sync` since they are local.